programs/ssl/ssl_client2.c

```diff
@@ -734,6 +734,8 @@ int main( int argc, char *argv[] )
     mbedtls_memory_buffer_alloc_init( alloc_buf, sizeof(alloc_buf) );
 #endif
 
+    test_hooks_init( );
```

Comment:
There are two schools of thought:

1. `test_hooks_xxx` does whatever it can, which happens to be nothing if `MBEDTLS_TEST_HOOKS` is disabled.
2. `test_hooks_xxx` is conditional to `MBEDTLS_TEST_HOOKS`.

I don't particularly care here, so I switched from (1) to (2).

It's possible that `test_hooks_xxx` could expand to cover things that don't require `MBEDTLS_TEST_HOOKS` (for example, things where it can work through platform functions). But we can change how the functions are called if and when that happens.

Reply:
I completely agree. I'm not opposed to (1) in general indeed there are some situations in which that would be clearer however in this situation with only one reference to test hooks I prefer (2). Thanks for making the change.